### DISCUSSION TITLE
Is it possible to setRotationCenter on the initialization of a shape?

### DISCUSSION THREAD

**Discussion by matthewtboyle** [geoman-io/leaflet-geoman#1453] [Questions & Help] [answered]
Hello,

In my program, I am creating various shapes programmatically that act as visual indicators for markers. I have created my own toolbar outside of geoman that will draw these shapes for the user. They are also able to use my custom menu to do things like change the length, width, and rotation.

The users do not have access to DrawMode, EditMode, or Rotate Mode, and I do not want my users to be able to interact with shapes with the leaflet controls that are still available to them ( These controls are to be used exclusively with markers only ).

My issue is, I want to be able to use { pmIgnore: true } in my options when creating a shape, but it seems that I can only use shape.setRotationCenter( latLng ) after the item as been created. 

My questions:
1 - Is there a way to set the rotation center when the shape is being created? If not, is this an option that could be added to available options when creating a shape?
2 - What is currently the most 'correct' way to dynamically set a geoman item to { pmIgnore: true }? I know I can use L.PM.optin to go from true to false. But what about false to true? 

Please let me know if you need more information, and I appreciate all of the work you have done for this great product!

**Comment by Falke-Design** [comment, marked as answer]
> 1 - Is there a way to set the rotation center when the shape is being created? If not, is this an option that could be added to available options when creating a shape?

if you creating the shape with `pmIgnore: true` or `OptIn` then the layer will not get the `pm` property. This also means that the can't be rotate too. So I don't really understand how you rotate a layer if it has not the `pm` property of Geoman.

> 2 - What is currently the most 'correct' way to dynamically set a geoman item to { pmIgnore: true }? I know I can use L.PM.optin to go from true to false. But what about false to true?

Just update the options of a layer `layer.options.pmIgnore = true`

I do not understand your workflow but a solution could be that you draw the layer and listening for `pm:create` on the `map`. If the event is fired, you set the rotation center and the apply `pmIgnore: true`

```
map.on('pm:create', (e)=>{
   e.layer.pm.setRotationCenter(LATLNG);
   e.layer.options.pmIgnore = true;
});
```
